src/dpms/dpmsWidget.tsx

```diff
@@ -116,9 +111,14 @@ const DpmsComponent = ({
   const [session, setSession] = useState(false);
   const [cluster, setCluster] = useState(false);
   const [entries, setEntries] = useState<string[]>([]);
+  const [databaseNames, setDatabaseNames] = useState<string[]>([]);
   const [columnResponse, setColumnResponse] = useState<string[]>([]);
   const [databaseDetails, setDatabaseDetails] = useState({});
   const [tableDescription, setTableDescription] = useState({});
+  const [apiError, setApiError] = useState(false);
+  const [schemaError, setSchemaError] = useState(false);
+  const [totalDatabase, setTotalDatabase] = useState<number>(0);
```

Comment:
let's be consistent and use plural for totalDatabases and totalTables

Reply:
also do we need this?  could we just derive this from databaseNames.length?

Reply:
The tree library that we are using renders initally once the inital data is loaded. So we need to check and render table only if all data is loaded. For that purpose we need total database state variable to check out of the function as well.